lib/pry/pry_class.rb

```diff
@@ -257,6 +257,8 @@ def self.run_command(command_string, options={})
 
     pry = Pry.new(:output   => output, :target   => target, :commands => options[:commands])
     pry.eval command_string
+    result = Pry.current[:pry_cmd_result].retval
+    result == Pry::Command::VOID_VALUE ? nil : result
```

Comment:
Hmm, specs all pass, but is it reasonable to expect :pry_cmd_result to always have a Result in it? Alternate option:

```ruby
result = Pry.current[:pry_cmd_result]
result && result.retval != Pry::Command::VOID_VALUE ? result.retval : nil
```
specs pass with alternate version as well.